cognite/powerops/client/data_classes/shop_results_compare.py

```diff
@@ -38,9 +47,221 @@ def plot_time_series(
         ax.legend()
         plt.show()
 
-    def yaml_difference(
+    def yaml_deep_diff(
         self,
-        yaml_1: ShopYamlFile,
-        yaml_2: ShopYamlFile,
+        post_run_yaml_1: ShopYamlFile,
+        post_run_yaml_2: ShopYamlFile,
+        **deepdiff_kwargs,
+    ) -> DeepDiff:
+        """DeepDiff wrapper for ShopYamlFile objects."""
+        return DeepDiff(
+            post_run_yaml_1.data,
+            post_run_yaml_2.data,
+            ignore_type_in_groups=[(int, float)],
+            **deepdiff_kwargs,
+        )
+
+    def yaml_difference_md(
+        self,
+        post_run_yaml_a: Optional[ShopYamlFile] = None,
+        post_run_yaml_b: Optional[ShopYamlFile] = None,
+        name_a: str = "Result A",
+        name_b: str = "Result B",
+        **deepdiff_kwargs,
+    ):
+        """
+        Returns a markdown string of the difference between two post run yaml files
+        In a notebook, this can be displayed using
+        ```
+        from IPython.display import Markdown
+
+        Markdown(yaml_difference_md(...))
+        ```
+        """
+        if post_run_yaml_a is None or post_run_yaml_b is None:
+            logger.error("Must provide two post run yaml files ")
+            return ""
+
+        yaml_deep_diff = self.yaml_deep_diff(post_run_yaml_a, post_run_yaml_b)
+        builder = YamlDiffMDBuilder(
+            yaml_deep_diff, post_run_yaml_a.data, post_run_yaml_b.data, name_a, name_b, **deepdiff_kwargs
+        )
+
+        output_string = f"# Changes from {name_a} to {name_b}"
+        output_string += "\n\n"
+        output_string += builder.build_md()
+        return output_string
+
+
+class YamlDiffMDBuilder:
+    def __init__(
+        self,
+        deep_diff: DeepDiff,
+        yaml_dict_a: dict,
+        yaml_dict_b: dict,
+        name_a: str,
+        name_b: str,
+    ):
+        self.deep_diff = deep_diff
+        self.yaml_dict_a = yaml_dict_a
+        self.yaml_dict_b = yaml_dict_b
+        self.name_a = name_a
+        self.name_b = name_b
+        self.md_string = None
```

Comment:
Overall very good design of this class, easy to follow the details of what you are doing at the different abstraction levels. 

I have one performance optimization suggestion that might not matter, but doing `self.md_string += "new string"` is expensive in Python as you are continuously creating a new string evetime you do it. The most efficient way is to make `self.md_string = []` a list append all the substrings to that list and do a `"".join(self.md_string)` at the end. 

Ref https://stackoverflow.com/a/36556380